lib/azuremetadata/azuremetadata.py

```diff
@@ -67,22 +66,18 @@ def get_disk_tag(self, device=None):
 
     @staticmethod
     def _find_root_device():
-        hex_device_id = os.stat("/").st_dev
-        root_device_id = "{}:{}".format(os.major(hex_device_id), os.minor(hex_device_id))
-
-        devices = glob.glob("/sys/block/*")
-        for device_path in devices:
-            device = os.path.basename(device_path)
-            partitions = glob.glob("/sys/block/{}/{}*".format(device, device))
-
-            for partition in partitions:
-                with open("{}/dev".format(partition), "r") as fh:
-                    device_id = fh.read().strip()
-
-                if device_id == root_device_id:
-                    return "/dev/{}".format(device)
+        """Returns detected root device path or None if detection failed."""
+        proc = subprocess.Popen(
+            ["findmnt", "-f", "-n", "-o", "SOURCE", "/"],
```

Comment:
You might want to think about:

```
-v, --nofsroot
              Do not print a [/dir] in the SOURCE column for bind mounts or btrfs subvolumes.
```

as -f will stop on first match we should make sure it does not stop on a subvolume. But it should be tested on a btrfs mounted system first

Reply:
Other than that I like the long versions of options simply because that makes reading the code easier :)

```
findmnt --first-only --noheadings --output=SOURCE
```

Reply:
> You might want to think about:
> 
> ```
> -v, --nofsroot
>               Do not print a [/dir] in the SOURCE column for bind mounts or btrfs subvolumes.
> ```
> 
> as -f will stop on first match we should make sure it does not stop on a subvolume. But it should be tested on a btrfs mounted system first

Good suggestion, thanks! Here's the output from a system with a btrfs root:

```
e127:~ # findmnt -n -o SOURCE /
/dev/sda2[/@/.snapshots/1/snapshot]
```

With `--nofsroot`:

```
e127:~ # findmnt -n -o SOURCE --nofsroot /
/dev/sda2
```

I'll add it.

Reply:
Added; switched to long option names as well.